test/react_client/event_helpers_test.dart

```diff
@@ -1502,5 +1649,530 @@ main() {
+          final mountNode = Element.div();
+          final content = div({'onDrag': (e) => event = e, 'draggable': true}, []);
+          render(content, mountNode);
+          return mountNode.children.single;
+        }
+
+        group('detects an anonymous JS object correctly', () {
+          test('when all fields are filled out', () {
+            final files = [File([], 'name1'), File([], 'name2'), File([], 'name3')];
+
+            final eventData = {
+              'dataTransfer': {
+                'files': files,
+                'types': ['d', 'e', 'f'],
+                'effectAllowed': 'none',
+                'dropEffect': 'move',
+              }
+            };
+
+            final node = renderAndGetRootNode();
+            Simulate.drag(node, eventData);
+
+            final dataTransfer = event.dataTransfer;
+
+            expect(dataTransfer, isNotNull);
+
+            final fileNames = dataTransfer.files.map((file) => (file as File).name);
+
+            expect(fileNames, containsAll(['name1', 'name2', 'name3']));
+            expect(dataTransfer.types, containsAll(['d', 'e', 'f']));
+            expect(dataTransfer.effectAllowed, 'none');
+            expect(dataTransfer.dropEffect, 'move');
+          });
+
+          test('when none of the fields are filled out', () {
+            final eventData = {'dataTransfer': {}};
+
+            final node = renderAndGetRootNode();
+            Simulate.drag(node, eventData);
+
+            final dataTransfer = event.dataTransfer;
+
+            expect(dataTransfer, isNotNull);
+            expect(dataTransfer.files, isNotNull);
+            expect(dataTransfer.files, isEmpty);
+            expect(dataTransfer.types, isNotNull);
+            expect(dataTransfer.types, isEmpty);
+            expect(dataTransfer.effectAllowed, null);
+            expect(dataTransfer.dropEffect, null);
+          });
+        });
+
+        group('detects a DataTransfer object correctly', () {
+          test('when none of the fields are filled out', () {
+            final eventData = {
+              'dataTransfer': DataTransfer(),
+            };
+
+            final node = renderAndGetRootNode();
+            Simulate.drag(node, eventData);
+
+            final dataTransfer = event.dataTransfer;
+
+            expect(dataTransfer, isNotNull);
+            expect(dataTransfer.files, isNotNull);
+            expect(dataTransfer.files, isEmpty);
+            expect(dataTransfer.types, isNotNull);
+            expect(dataTransfer.types, isEmpty);
+            expect(dataTransfer.effectAllowed, 'none');
+            expect(dataTransfer.dropEffect, 'none');
+          });
+        });
+      });
+    });
   });
 }
+
+// ignore: avoid_implementing_value_types
+class MockKeyboardEvent extends Mock implements KeyboardEvent {}
+
+// ignore: avoid_implementing_value_types
+class MockMouseEvent extends Mock implements MouseEvent {}
+
+enum SyntheticEventType {
+  SyntheticClipboardEvent,
```

Comment:
#nit enum values should be lowercase (I see why they aren't here, but I think it'd be more readable if they matched the convention)